src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.cs

```diff
@@ -1117,6 +1117,60 @@ public AssemblyReference GetAssemblyReference(AssemblyReferenceHandle handle)
             return new AssemblyReference(this, handle.Value);
         }
 
+        internal AssemblyName GetAssemblyName()
+        {
+            StringHandle nameHandle = AssemblyTable.GetName();
+
+            string name = (!nameHandle.IsNil) ? GetString(nameHandle) : null;
+
+            Version version = AssemblyTable.GetVersion();
+
+            StringHandle cultureHandle = AssemblyTable.GetCulture();
+            string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : null;
+
+            BlobHandle publicKeyHandle = AssemblyTable.GetPublicKey();
+            byte[] publicKeyOrToken = !publicKeyHandle.IsNil ? GetBlobBytes(publicKeyHandle) : null;
+
+            AssemblyHashAlgorithm assemblyHashAlgorithm = AssemblyTable.GetHashAlgorithm();
+
+            AssemblyFlags flags = AssemblyTable.GetFlags();
+
+            var assemblyName = new AssemblyName(name)
+            {
+                Version = version,
+                CultureName = cultureName,
+                HashAlgorithm = (Configuration.Assemblies.AssemblyHashAlgorithm)assemblyHashAlgorithm
+            };
+            assemblyName.SetPublicKey(publicKeyOrToken);
+            SetFlags(assemblyName, flags);
+
+            return assemblyName;
+        }
+
+        private void SetFlags(AssemblyName assemblyName, AssemblyFlags flags)
+        {
+            if ((flags & AssemblyFlags.PublicKey) != 0)
+                assemblyName.Flags |= AssemblyNameFlags.PublicKey;
+            else
+                assemblyName.Flags &= AssemblyNameFlags.PublicKey;
+
+            if ((flags & AssemblyFlags.Retargetable) != 0)
+                assemblyName.Flags |= AssemblyNameFlags.Retargetable;
+            else
+                assemblyName.Flags &= AssemblyNameFlags.Retargetable;
```

Comment:
This doesn't look right. Won't it clear the other flags when the else branch is taken?